source/articles/2015/save-money-and-be-happier-by-updating-your-gems-every-monday-morning.md

```diff
@@ -0,0 +1,256 @@
+---
+title: Save Money and Be Happier by Updating Your Gems Every Monday Morning
+description: An approach to avoiding the technical debt of gem drift
+
+author: Adarsh Pandit
+category: writing
+date: 2015-11-12
+logo: ruby_on_rails_logo.png
+published: true
+tags: [ruby on rails, update gems, versions, technical debt, upgrade gems]
+---
+
+As a newcomer to Rails,
+understanding Gems and the Gemfile
+is fairly straightforward:
+You include a gem that you need
+and you get it included everywhere.
+
+We don't often think about it,
+but that last bit is important:
+we are _literally including_
+all of the code from each gem
+into our application, all the time.
+
+There are other implications of
+this `include` behavior which should be considered,
+like speed
+but here I'm going to address one:
+Gem Drift and the associated cost.
+
+## An Example
+
+Let's say I include an auth gem called `huge_bouncer`
+in my app to do authentication and authorization.
+(Note: this is not a real gem, but a name I love.
+Feel free to use it but please attribute here
+so I can feel good about my gem-naming skills.)
+
+Over time, the nice person who wrote `huge_bouncer`
+reviews issues and PRs at night and on weekends
+and keeps development moving at a good pace.
+She is a saint and deserves an award
+or at least you should buy her a beer
+if you run into her at conference.
+
+Newly released gem versions are, for the most part,
+new features and security patches
+(particularly in auth libraries).
+
+Not keeping up to date with new features
```

Comment:
Good idea. Did in 2103219. Also made "up-to-date" three words connected by dashes.